base/float.jl

```diff
@@ -204,23 +303,31 @@ trunc(::Type{Unsigned}, x::Float32) = trunc(UInt,x)
 trunc(::Type{Unsigned}, x::Float64) = trunc(UInt,x)
 trunc(::Type{Integer}, x::Float32) = trunc(Int,x)
 trunc(::Type{Integer}, x::Float64) = trunc(Int,x)
+trunc{T<:Integer}(::Type{T}, x::Float16) = trunc(T, Float32(x))
 
 # fallbacks
 floor{T<:Integer}(::Type{T}, x::AbstractFloat) = trunc(T,floor(x))
+floor{T<:Integer}(::Type{T}, x::Float16) = floor(T, Float32(x))
```

Comment:
We could remove most of these definitions if we re-enable support for `Float16` intrinsics (which was disabled back in 2013 71ae4931e)